src/tours/Tour/index.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useState } from 'react';
+import Tour from 'reactour';
+import {
+    getGroupComparisonSteps,
+    getVirtualStudySteps,
+    groupComparisonId,
+    virtualStudyId,
+} from '../Steps';
+import { TourProps, TourMapProps } from './types';
+import './styles.scss';
+
+export const setTourLocalStorage = (id: string, value: string) => {
+    localStorage.setItem('web-tour', id);
+    localStorage.setItem(id, value);
+};
+
+export const setTourLocalStorageFromURL = () => {
+    /**
+     * If the url contains a query param 'webtour', set the localStorage
+     */
+    const urlStr = window.location.href.split('?')[1];
+    if (urlStr) {
+        const urlSearchParams = new URLSearchParams(urlStr);
+        const webTour = urlSearchParams.get('webtour');
+        if (webTour && [groupComparisonId, virtualStudyId].includes(webTour)) {
+            setTourLocalStorage(webTour, '0');
+        }
+    }
+};
+
+export default function WebTour({
+    hideEntry = true,
+    isLoggedIn = false,
+    studies = 0,
+}: TourProps) {
+    const [currentTour, setCurrentTour] = useState<string | null | undefined>(
+        null
+    );
+    const [isOpen, setIsOpen] = useState(true);
+    const [gotoStep, setGotoStep] = useState(null);
+    const [lockTour, setLockTour] = useState(false);
+
+    const [startAt, setStartAt] = useState<number>(0);
+    const endTour = () => setIsOpen(false);
+    const endTourWithBtn = (e: any) => {
+        e.preventDefault();
```

Comment:
are you sure you need these?

Reply:
I think so, `e.preventDefault(); e.stopPropagation();` is required in the click event here. 
Without it, it would report `TypeError: Cannot read properties of null (reading 'getBoundingClientRect')` when click 'Skip All' button on the second step of virtual study tour.